py-rattler/rattler/networking/client.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+from rattler.rattler import PyClientWithMiddleware
+from rattler.networking.middleware import AuthenticationMiddleware, MirrorMiddleware
+
+
+class Client:
+    """
+    A client that can be used to make authenticated requests.
+    """
+
+    def __init__(self, middlewares: list[MirrorMiddleware] | None = None) -> None:
+        self._client = PyClientWithMiddleware([middleware._middleware for middleware in middlewares] if middlewares else None)
+
+    @classmethod
+    def _from_ffi_object(cls, client: PyClientWithMiddleware) -> Client:
+        """
+        Construct py-rattler Client from PyClientWithMiddleware FFI object.
+        """
+        client = cls.__new__(cls)
+        client._client = client
+        return client
+
+    def __repr__(self) -> str:
+        """
+        Returns a representation of the Client
+
+        Examples
+        --------
+        ```python
+        >>> Client()
+        Client()
+        >>>
+        ```
+        """
+        return f"{type(self).__name__}()"
+
+
+class AuthenticatedClient(Client):
```

Comment:
not sure whether we should still keep `AuthenticatedClient` now that we have `Client`. Any opinions? Since it's not really much code i would be inclined to keeping it.

Reply:
Id prefer to only keep `Client`. Im not too well versed in python common practices but instead of subclasses we could also add a static method to `Client` to initialize with the authentication middleware?

Im fine either way.

Reply:
yeah, let's create a static method instead. this will be a breaking change, can you take care of adding a note in the changelog afterwards?